src/components/apps/launch/ResourceRequirements.js

```diff
@@ -0,0 +1,357 @@
+/**
+ * @author psarando
+ *
+ * Form fields for adjusting (or reviewing) a step's resources requirements.
+ * The range of each requirement is determined either by
+ * the given min and max values provided in the `requirements` prop,
+ * or from 0 to the default max prop (from client configs),
+ * or from 0 to a reasonable max value set by this component.
+ */
+import React from "react";
+
+import { FastField } from "formik";
+import numeral from "numeral";
+
+import ids from "./ids";
+import messages from "./messages";
+import styles from "./styles";
+
+import {
+    FormSelectField,
+    build as buildDebugId,
+    getMessage,
+    withI18N,
+} from "@cyverse-de/ui-lib";
+
+import {
+    makeStyles,
+    ExpansionPanel,
+    ExpansionPanelSummary,
+    ExpansionPanelDetails,
+    MenuItem,
+    Paper,
+    Table,
+    TableBody,
+    TableCell,
+    TableContainer,
+    TableRow,
+    Typography,
+} from "@material-ui/core";
+
+import { ExpandMore } from "@material-ui/icons";
+
+const useStyles = makeStyles(styles);
+
+const ONE_GB = 1024 * 1024 * 1024;
```

Comment:
Maybe this could be refactored as a global util constant (it's reused at least in the app launch stories).